BinaryVault/contracts/interfaces/IBinaryVault.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IBinaryVault {
```

Comment:
Please order functions layout by type
Refer https://docs.soliditylang.org/en/v0.8.17/style-guide.html#order-of-layout

Recommendation Order:
State Variables
Events
Modifiers
Constructor
External functions
External functions that are view
External functions that are pure
Public functions
Internal functions
Private functions